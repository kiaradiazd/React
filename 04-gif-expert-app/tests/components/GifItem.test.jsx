import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"

describe('pruebas es <GifItem />', () => {

    const title = 'UdemyReact';
    const url = 'https://www.udemy.com/course/react-cero-experto/learn/lecture/32021266#questions';

    test('Debe de hacer match con el snapshot', () => {


        const { container } = render( <GifItem title={ title } url={ url }/>);
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el url y el alt indicado', () => {
      
        render( <GifItem title={ title } url={ url }/> );
        screen.debug();

        expect( screen.getByRole('img').alt).toBe( title );
        expect( screen.getByRole('img').src).toBe( url )

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('Debe de mostrar el titulo en el componente', () => {
      
        render( <GifItem title={ title } url={ url }/> );
        expect( screen.getByText( title ) ).toBeTruthy(); 
    });
    
    
});