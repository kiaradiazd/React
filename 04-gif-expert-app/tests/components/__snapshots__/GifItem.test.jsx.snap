// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas es <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="UdemyReact"
      src="https://www.udemy.com/course/react-cero-experto/learn/lecture/32021266#questions"
    />
    <p>
      UdemyReact
    </p>
  </div>
</div>
`;
